### ISSUE TITLE
[tune] meaning of each column of `analysis.dataframe()`?

### ISSUE THREAD

**Issue by rightx2** [ray-project/ray#9351]
I I have a code to run a simple neural network total 10 epochs and use `ASHAScheduler` for hp search and got a result dataframe from `analysis.dataframe()`:

<img width="835" alt="Screen Shot 2020-07-08 at 14 17 42" src="https://user-images.githubusercontent.com/10606994/86879826-ba7e9f80-c126-11ea-89a8-c1b416780baa.png">

Unfortunately, I have a problem analyzing these results:  

1. I'd like to know what the `done` column means in this case. It looks contradictory because experiments whose `done` value are `False` have `training_iteration` of 10, which means it run full 10 epochs? Have no idea why it got `False` though it run full epochs.

2. what's meaning of `since_restore`? There are some columns having `_since_restore` as suffix but wonder what it means. I can't find info about it in the doc page, either. What's the meaning of `restore`? restoring from what?

3. Lastly, the reason that there are some experiments with ONLY 1 iteration is related to how `ASHAScheduler` works?

I wish there is a document of analyzing the result dataframe as well!

Thanks, in advance.

**Comment by ioannisa92** [comment]
I have the same question!

**Comment by krfricke** [comment]
Hi,

1. I'll have to look into the code for this, but my guess is that stopping a run triggers setting of the `done` function, while stopping the whole experiment does not do this.

2. This refers to checkpoints. Checkpoints are stored periodically (either in memory or on disk) and are needed for some scheduling algorithms, like PBT, or when you need to share sparse resources between different runs. The `_since_restore` suffixes count metrics since the last restore of a checkpoint to  see if the performance develops differently after restoring from a checkpoint. In PBT, runs restore from checkpoints of different runs, so it's quite informative there.

3. Yes, ASHA stops bad performing trials early and continues to run only some of them. You can set the `grace_period=n` parameter to make sure each trial runs for at least `n` steps. You can also try to reduce the `reduction_factor` to something like 2 to make the pruning less aggressive. Lastly, you should also pass `max_t=m` where `m` is the total number of times you report your metrics back to Tune. E.g. if you train for 10 epochs and report a validation loss at the end of each epoch, set `max_t=10`

**Comment by krfricke** [comment]
Sorry for getting back so late. Apparently we only log `done=True` in CSVs if the trial explicitly passes this in `tune.report()` - i.e. it is treated like any other metric. But since we update`done` internally, too, this is confusing. I added an issue to track this at #12037. 

I'll close this issue for now, but please let me know if there's other questions you have regarding the results dataframe.
